dockstore-webservice/src/main/java/io/dockstore/webservice/resources/CollectionResource.java

```diff
@@ -223,6 +225,12 @@ private void addCollectionEntriesToCollection(Collection collection) {
         collectionEntries.addAll(collectionWorkflowsWithVersions);
         collectionEntries.addAll(collectionServicesWithVersions);
         collectionEntries.addAll(collectionToolsWithVersions);
+        collectionEntries.forEach(entry -> {
+            //Object entryDAO;
+            List<Label> labels = workflowDAO.getLabelByEntryId(entry.getId());
+            List<String> labelStrings = labels.stream().map(Label::getValue).collect(Collectors.toList());
```

Comment:
I haven't looked at the schema too closely, but I believe you could just query the value directly from the DB.

Reply:
The db has an `entry_label(s)` table, where the first column is `entryid`, second is `labelid`, I think in this case, I'm still doing the same thing, or rather it's more work this way since I will then need to match id with value

Reply:
I could be wrong, but I think you can do:

`@NamedQuery(name = "io.dockstore.webservice.core.Entry.findLabelByEntryId", query = "SELECT e.labels.value FROM Entry e WHERE e.id = :entryId")`, then you don't have to worry about the JOIN table.

Reply:
Pretty sure my comment won't work; ignore.